xknx/devices/climate.py

```diff
@@ -164,63 +208,74 @@ def is_on(self):
-        if not self.setpoint_shift.initialized:
+        if not self._setpoint_shift.initialized:
             return False
-        if self.setpoint_shift.value is None:
+        if self.setpoint_shift is None:
             return False
         if not self.target_temperature.initialized:
             return False
         if self.target_temperature.value is None:
             return False
         return True
 
-    async def set_target_temperature(self, target_temperature):
-        """Calculate setpoint shift shift and send it to  KNX bus."""
+    @property
+    def temperature_step(self):
+        """Return smallest possible temperature step."""
         if self.initialized_for_setpoint_shift_calculations:
-            await self.set_target_temperature_setpoint_shift(target_temperature)
-        # broadcast new target temperature and set internally
-        await self.target_temperature.set(target_temperature)
+            return self._setpoint_shift.setpoint_shift_step
+        return 0.1
 
-    async def set_target_temperature_setpoint_shift(self, target_temperature):
-        """Set target temperature via setpoint_shift group address."""
-        temperature_delta = target_temperature-self.target_temperature.value
-        setpoint_shift_delta = int(temperature_delta/self.setpoint_shift_step)
-        setpoint_shift = self.setpoint_shift.value + setpoint_shift_delta
+    async def set_target_temperature(self, target_temperature):
+        """Send new target temperature or setpoint_shift to KNX bus."""
+        if self.initialized_for_setpoint_shift_calculations:
+            temperature_delta = target_temperature-self.base_temperature
+            await self.set_setpoint_shift(temperature_delta)
+        else:
+            await self.target_temperature.set(target_temperature)
 
-        if setpoint_shift > self.setpoint_shift_max:
-            raise DeviceIllegalValue("setpoint_shift_max exceeded", setpoint_shift)
-        elif setpoint_shift < self.setpoint_shift_min:
-            raise DeviceIllegalValue("setpoint_shift_min exceeded", setpoint_shift)
+    @property
+    def base_temperature(self):
+        """Return the base temperature."""
+        if self.initialized_for_setpoint_shift_calculations:
+            return self.target_temperature.value - self.setpoint_shift
+        return None
 
-        await self.setpoint_shift.set(setpoint_shift)
+    @property
+    def setpoint_shift(self):
+        """Return current offset from base temperature in Kelvin."""
+        return self._setpoint_shift.value
+
+    async def set_setpoint_shift(self, offset):
+        """Send new temperature offset to KNX bus."""
+        base_temperature = self.base_temperature
+        await self._setpoint_shift.set(offset)
+        # broadcast new target temperature and set internally
+        if base_temperature is not None:
+            await self.target_temperature.set(base_temperature + self.setpoint_shift)
```

Comment:
this yields 
`[xknx.log] Error while processing telegram <ConversionError description="dst_add not set" />`
when using group_address_target_temperature_state and no  group_address_target_temperature.

Reply:
The actor will send an update if the value changes thus I don't think setting it there is neccessary

Reply:
In my system the update from the bus is send immediately, so I tend to say an update is not necessary. 

But if we do so, we should only try to broadcast the target temperature if a set address is set.

Reply:
I was thinking about defining a `writeable` property in RemoteValue() tocheck this.

Reply:
I like the idea of the `writeable` property. 

On the other hand it makes the code more complex. Shouldn't this generally be the case? Can we not just check in `RemoteValue.set()` if the there is a GA defined for writing ?

Reply:
I like the idea of checking for writable in RemoteValue.set(). I would log a warning instead of an info if it is called for a non-writable device. 
To suppress this warning I'd still check for writable in climate.py.